src/diff/index.js

```diff
@@ -311,11 +311,11 @@ export function unmount(vnode, ancestorComponent) {
 		}
 
 		r.base = r._parentDom = null;
-		if (r = r._prevVNode) unmount(r, ancestorComponent);
+		if (r = r._prevVNode) unmount(r, ancestorComponent, skipRemove);
 	}
 	else if (r = vnode._children) {
 		for (let i = 0; i < r.length; i++) {
-			unmount(r[i], ancestorComponent);
+			unmount(r[i], ancestorComponent, skipRemove && vnode.type!==Fragment);
```

Comment:
The for `Fragment` is necessary, because `_dom` will point to the first child.

Reply:
In the case of a fragment, we could skip removing `_dom` and set `skipRemove=false`.

Reply:
Noticed that this isn't only true for `Fragments` but also for HOC-Components. They may have different `_dom` and `_lastDomChild` pointers. Went with checking for the latter property and only removing the dom when it is `null`.